Userland/Libraries/LibDeviceTree/DeviceTree.h

```diff
@@ -0,0 +1,191 @@
+/*
+ * Copyright (c) 2024, Leon Albrecht <leon.a@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/Concepts.h>
+#include <AK/Endian.h>
+#include <AK/Function.h>
+#include <AK/HashMap.h>
+#include <AK/IterationDecision.h>
+#include <AK/MemoryStream.h>
+#include <AK/Optional.h>
+#include <AK/Span.h>
+
+namespace DeviceTree {
+
+struct DeviceTreeProperty {
+    ReadonlyBytes raw_data;
+
+    size_t size() const { return raw_data.size(); }
+
+    StringView as_string() const { return StringView(raw_data.data(), raw_data.size() - 1); }
+    Vector<StringView> as_strings() const { return as_string().split_view('\0'); }
+
+    // Note: as<T> does not convert endianness, so all structures passed in
+    //       should use BigEndian<T>s for their members and keep ordering in mind
+    // Note: The Integral variant does convert endianness, so no need to pass in BigEndian<T>s
+    template<typename T>
+    T as() const
+    {
+        VERIFY(raw_data.size() == sizeof(T));
+        T value;
+        __builtin_memcpy(&value, raw_data.data(), sizeof(T));
+        return value;
+    }
+
+    template<typename T>
+    requires(alignof(T) <= 4 && !IsIntegral<T>)
```

Comment:
Why `alignof` and not `sizeof` here?

Reply:
because this is handing out a reference into the device tree, which only guarantees alignment to 4 bytes for properties
so anything with higher llignment requirements needs to be copied out, instead of just being a reference

Reply:
Ah ok, this seems fine then